amazon-kinesis-client/src/test/java/software/amazon/kinesis/coordinator/SchedulerTest.java

```diff
@@ -1080,6 +1097,87 @@ private void mockListLeases(List<StreamConfig> configs) throws ProvisionedThroug
                         .shardId("some_random_shard_id")).collect(Collectors.toList()));
     }
 
+    @Test
+    public void testStreamConfigsArePopulatedWithStreamArnsInMultiStreamMode() {
+        final String streamArnStr = constructStreamArnStr(TEST_REGION, 111122223333L, "some-stream-name");
+        when(multiStreamTracker.streamConfigList()).thenReturn(Stream.of(
+                        // Each of scheduler's currentStreamConfigMap entries should have a streamARN in
+                        // multi-stream mode, regardless of whether the streamTracker-provided streamIdentifiers
+                        // were created using serialization or stream ARN.
+                        StreamIdentifier.multiStreamInstance(constructStreamIdentifierSer(TEST_ACCOUNT, streamName)),
+                        StreamIdentifier.multiStreamInstance(Arn.fromString(streamArnStr), TEST_EPOCH)
+                )
+                .map(streamIdentifier -> new StreamConfig(streamIdentifier, TEST_INITIAL_POSITION))
+                .collect(Collectors.toList()));
+        retrievalConfig = new RetrievalConfig(kinesisClient, multiStreamTracker, applicationName)
+                .retrievalFactory(retrievalFactory);
+        scheduler = new Scheduler(checkpointConfig, coordinatorConfig, leaseManagementConfig, lifecycleConfig,
+                metricsConfig, processorConfig, retrievalConfig);
+
+        final Set<String> expectedStreamArns =
+                Sets.newHashSet(constructStreamArnStr(TEST_REGION, TEST_ACCOUNT, streamName), streamArnStr);
+
+        final Set<String> actualStreamArns = scheduler.currentStreamConfigMap().values().stream()
+                .map(sc -> sc.streamIdentifier().streamArnOptional().orElseThrow(IllegalStateException::new).toString())
+                .collect(Collectors.toSet());
+
+        assertEquals(expectedStreamArns, actualStreamArns);
+    }
+
+    @Test
+    public void testOrphanStreamConfigIsPopulatedWithArn() {
+        final String streamIdentifierSerializationForOrphan = constructStreamIdentifierSer(TEST_ACCOUNT, streamName);
+        assertFalse(multiStreamTracker.streamConfigList().stream()
+                .map(sc -> sc.streamIdentifier().serialize())
+                .collect(Collectors.toSet())
+                .contains(streamIdentifierSerializationForOrphan));
+
+        when(leaseCoordinator.getCurrentAssignments()).thenReturn(Collections.singletonList(
+                new ShardInfo(TEST_SHARD_ID, null, null, null, streamIdentifierSerializationForOrphan)));
+        retrievalConfig = new RetrievalConfig(kinesisClient, multiStreamTracker, applicationName)
+                .retrievalFactory(retrievalFactory);
+        scheduler = new Scheduler(checkpointConfig, coordinatorConfig, leaseManagementConfig, lifecycleConfig,
+                metricsConfig, processorConfig, retrievalConfig);
+
+        final Scheduler schedulerSpy = spy(scheduler);
```

Comment:
Why do you need a spy? can you not invoke runProcessLoop on scheduler?

Reply:
Good callout! Spy is not needed here; must've copied from a previous unit test and forgotten to clean it up.